### PR TITLE
feat:info verb changes

### LINKED ISSUE #1356: APKAM - info verb to include APKAM namespace access details
### Is your feature request related to a problem? Please describe.

info verb to include APKAM namespace access details
https://github.com/atsign-foundation/at_protocol/blob/trunk/decisions/2023-01-pkam-per-app-and-device.md

### Describe the solution you'd like

Modify info verb implementation to include APKAM namespace details

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by purnimavenkatasubbu**
**- What I did**
- Added enroll namespace details in the info_verb_handler
**- How I did it**
- check whether enrollId is not null, connection is authenticated and fetch the details of key $enrollId.$newEnrollmentKeyPattern.$enrollManageNamespace$atSign from keystore and add to info verb response 

**- How to verify it**
- run the info verb functional tests to validate regression
- functional test added and commented. Can be uncommented along with enroll changes PR

### CODE REVIEW

File: packages/at_secondary_server/lib/src/verb/handler/info_verb_handler.dart [hunk 1]
```diff
@@ -73,4 +92,14 @@ class InfoVerbHandler extends AbstractVerbHandler {
         "$uSeconds seconds";
     return uptimeAsWords;
   }
+
+  Future<String?> _getapkamMetadataKey(String? apkamMetadataKey) async {
+    AtData? result;
+    try {
+      result = await keyStore.get(apkamMetadataKey);
+    } on Exception catch (e) {
```
Comment (murali-shris) [inline packages/at_secondary_server/lib/src/verb/handler/info_verb_handler.dart]: catch specific exception (KeyNotFoundException)

File: packages/at_secondary_server/lib/src/verb/handler/info_verb_handler.dart [hunk 2]
```diff
@@ -73,4 +92,14 @@ class InfoVerbHandler extends AbstractVerbHandler {
         "$uSeconds seconds";
     return uptimeAsWords;
   }
+
+  Future<String?> _getapkamMetadataKey(String? apkamMetadataKey) async {
```
Comment (murali-shris) [inline packages/at_secondary_server/lib/src/verb/handler/info_verb_handler.dart]: naming convention
_getApkamMetadataKey

**Review by murali-shris — CHANGES_REQUESTED** [review]
please address couple of review comments

**Review by murali-shris — DISMISSED** [review]

File: tests/at_functional_test/test/info_verb_test.dart [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  test('info verb test without authentication', () async {
+    await socket_writer(socketFirstAtsign!, 'info');
+    var infoVerbResponse = await read();
+    print('info verb response : $infoVerbResponse');
+    infoVerbResponse = infoVerbResponse.replaceAll('data:', '');
+    var infoResponse = jsonDecode(infoVerbResponse);
+    expect(infoResponse['version'], isNotEmpty);
+  });
+
+  // commenting the test as the server doesn't have enroll verb changes yet
```
Comment (gkc) [inline tests/at_functional_test/test/info_verb_test.dart:39]: Please use a check of server version to determine whether to run the test or not

**Review by gkc — CHANGES_REQUESTED** [review]

**Review by gkc — DISMISSED** [review]

**Comment by gkc** [comment]
@murali-shris functional tests are failing, all with apkam-related errors

**Comment by murali-shris** [comment]
> @murali-shris functional tests are failing, all with apkam-related errors

- First failure was related to environment. pkam private key was not available on server and all tests failed. passed on re-run
- second failure was related to a issue in the info verb functional test. Replaced the method call socketConnection(with security context) with secure_socket_connection(without security context)

**Review by murali-shris — APPROVED** [review]

**Comment by murali-shris** [comment]
> > @murali-shris functional tests are failing, all with apkam-related errors
> 
> * First failure was related to environment. pkam private key was not available on server and all tests failed. passed on re-run
> * second failure was related to a issue in the info verb functional test. Replaced the method call socketConnection(with security context) with secure_socket_connection(without security context)

@gkc I am merging this PR

**Comment by gkc** [comment]
> * First failure was related to environment. pkam private key was not available on server and all tests failed. passed on re-run

@murali-shris How come the private key was available on re-run?

**Comment by murali-shris** [comment]
> > * First failure was related to environment. pkam private key was not available on server and all tests failed. passed on re-run
> 
> @murali-shris How come the private key was available on re-run?

Attempt # 1 - https://github.com/atsign-foundation/at_server/actions/runs/5478087010/attempts/1
Load pkam keys script failed for alice🛠 and few atsigns due to connectivity issue to secondary.  Hence the pkam key was not updated before the test run
Attempt # 2 - https://github.com/atsign-foundation/at_server/actions/runs/5478087010/jobs/9978333657
No errors during load keys script and functional tests ran with few errors which was fixed in subsequent commit

**Comment by gkc** [comment]
Got it. Can you create a ticket please to have the job fail if the loadKeys step fails. And perhaps loadKeys should retry after N seconds if it fails

**Pull request merged**